command_parse.c

```diff
@@ -1175,6 +1177,41 @@ enum CommandResult parse_set(struct Buffer *buf, struct Buffer *s,
       query = true;
       s->dptr++;
     }
+    else if (*s->dptr == '+' || *s->dptr == '-')
+    {
+      if (prefix)
+      {
+        mutt_buffer_printf(err, "ERR04 can't use prefix when incrementing or "
+                                "decrementing a variable");
+        return MUTT_CMD_WARNING;
+      }
+
+      if (reset || unset || inv)
+      {
+        mutt_buffer_printf(err, "ERR05 can't set a variable with the '%s' command",
+                           set_commands[data]);
+        return MUTT_CMD_WARNING;
+      }
+
+      if (!(DTYPE(he->type) == DT_NUMBER || DTYPE(he->type) == DT_LONG))
+      {
+	mutt_buffer_printf(err, "Variable '%s' has an invalid type %d",
+			   buf->data, he->type);
+        return MUTT_CMD_WARNING;
+      }
```

Comment:
This works, but it's not the _best_ place.
If we add support to another type, e.g. `DT_SLIST`, we'd have to change the code here too.

The next step down the call-tree is the **subset** code.
Again, this doesn't really know about types, only inheritance.
(Note: there are no `DT_*` constants in that code)

One step deeper, we reach the **set** code.
This understands the _types_ of data.

I think we should check here, just before we'd crash :-)
We can return:
- `CSR_ERR_INVALID` - operation failed
- A new `CSR_INV_` flag (or'd together) meaning: operation not supported
- An error message